### ISSUE TITLE
Clean-up: Remove unused template support from snippets tool

### ISSUE THREAD

**Issue by goderbauer** [flutter/flutter#103581]
The code for API doc samples is no longer embedded in the doc comments. It now lives under `examples/api`. This allows us to simplify the [snippets](https://github.com/flutter/assets-for-api-docs/tree/master/packages/snippets) tool by removing support for templates from it. With the new way of hosting API samples under `examples/api`, templates are no longer used.

**Comment by goderbauer** [comment]
@gspencergoog Are the scripts `flutter_sample_editor.dart` and `extract_sample.dart` still useful now that sample code is no longer embedded in our API docs? Can those scripts also be used for snippet code, which is still embedded in the API docs?

**Comment by goderbauer** [comment]
I am deleting the unused templates from the flutter repository (not the support for using them in the tool) in https://github.com/flutter/flutter/pull/103575.

**Comment by gspencergoog** [comment]
No, they're not all that useful anymore.  The sample editor was intended for a world where we left the samples in the API docs, and the extraction tool was mainly to help with migration. We can just nuke them both.
